Dockerfile

```diff
@@ -0,0 +1,37 @@
+# Docker 17.05 or higher required for multi-stage builds
+FROM rust:1.26.0-stretch as builder
+ADD . /app
+WORKDIR /app
+# Make sure that this matches in .travis.yml
+ARG RUST_TOOLCHAIN=nightly
+RUN \
+    apt-get -qq update && \
+    apt-get -qq install -y default-libmysqlclient-dev && \
+    \
+    rustup default ${RUST_TOOLCHAIN} && \
+    cargo --version && \
+    rustc --version && \
+    mkdir -m 755 bin && \
+    cargo build --release && \
+    cp /app/target/release/service /app/bin
+
+
+FROM debian:stretch-slim
+# FROM debian:stretch  # for debugging docker build
+RUN \
+    groupadd --gid 10001 app && \
+    useradd --uid 10001 --gid 10001 --home /app --create-home app && \
+    \
+    apt-get -qq update && \
+    apt-get -qq install -y default-libmysqlclient-dev libssl-dev && \
+    rm -rf /var/lib/apt/lists
+
+COPY --from=builder /app/bin /app/bin
+
+WORKDIR /app
+USER app
+
+# override rocket's dev env defaulting to localhost
+ENV ROCKET_ADDRESS 0.0.0.0
```

Comment:
the 2 process should have different ports as well

Reply:
This can also be set in `Rocket.toml` fwiw. We already set the port in there, not sure which is preferable from an ops standpoint. (and we have #24 open for other settings that can go in there, like how many threads to run on)

Reply:
https://github.com/mozilla/fxa-email-service/blob/master/Rocket.toml

Reply:
(it can have separate sections for dev, stage and prod or whatever)

Reply:
it's probably easier for ops to configure anything via env variables so they don't have to configure a volume mount for a config file. of course, ROCKET_PORT is already supported as an env var